pkg/cmd/infra/router/router.go

```diff
@@ -221,19 +219,25 @@ func (o *RouterSelection) Complete() error {
 }
 
 // NewFactory initializes a factory that will watch the requested routes
-func (o *RouterSelection) NewFactory(routeclient routeclient.RoutesGetter, projectclient projectclient.ProjectResourceInterface, kc kclientset.Interface) *controllerfactory.RouterControllerFactory {
-	factory := controllerfactory.NewDefaultRouterControllerFactory(routeclient, kc)
+func (o *RouterSelection) NewFactory(routeClient routeclient.RoutesGetter, projectClient *projectclientset.Clientset, kc kclientset.Interface) *controllerfactory.RouterControllerFactory {
```

Comment:
Switching from a targeted interface to a concrete (broader) clientset is a code smell. What extra API surface is required here?